dio/lib/src/options.dart

```diff
@@ -99,7 +99,13 @@ mixin OptionsMixin {
   late String _baseUrl;
 
   set baseUrl(String value) {
-    assert(value.isEmpty || kIsWeb || Uri.parse(value).host.isNotEmpty);
+    if (value.isNotEmpty && !kIsWeb && Uri.parse(value).host.isEmpty) {
+      throw ArgumentError.value(
+        value,
+        'baseUrl',
+        'Must has a valid host on platforms other than the Web',
```

Comment:
```suggestion
        'Must be a valid URL on platforms other than Web.',
```

Reply:
I took other callers of the error as examples and they didn't seem to have the dot at the end of the message.

Reply:
Ah ok, I don't mind either way. Only remembered that you always added a dot at the end :)